src/nupic/utils/GroupBy.hpp

```diff
@@ -183,12 +182,29 @@ namespace nupic
   // ==========================================================================
 
   template<typename Iterator0, typename KeyFn0,
+           typename KeyType = typename std::result_of<
+             KeyFn0(decltype(*std::declval<Iterator0>()))
+               >::type>
+  static KeyType min_front_key(KeyType frontrunner,
```

Comment:
I went ahead and changed it to `minFrontKey`.

Should we start a big naming guidelines proposal for `group_by` and `iter_group_by`? This file already explains why it uses this unusual naming. It's really a "when in Rome" naming convention. C++ iterator/template library code usually looks like this. `groupBy` and `iterGroupBy` would feel weird.